pkg/parser/parser.go

```diff
@@ -3,19 +3,27 @@ package parser
 import (
 	"bytes"
 	_ "embed"
+	"encoding/json"
 	"log"
 	"text/template"
+
+	nbcontractv1 "github.com/Azure/agentbaker/pkg/proto/nbcontract/v1"
 )
 
 var (
 	//go:embed cse_cmd.sh.gtpl
 	bootstrapTrigger         string
-	bootstrapTriggerTemplate = template.Must(template.New("triggerBootstrapScript").Parse(bootstrapTrigger)) //nolint:gochecknoglobals
+	bootstrapTriggerTemplate = template.Must(template.New("triggerBootstrapScript").Funcs(funcMap).Parse(bootstrapTrigger)) //nolint:gochecknoglobals
+	//nolint:gochecknoglobals
+	funcMap = template.FuncMap{
```

Comment:
these do not need to be global vars, we can do something like this if we do want to use func map:
https://github.com/Azure/AgentBaker/blob/bb5ee284ab17e8b610a6e7fc77e08519c67c1532/pkg/agent/baker.go#L183

Reply:
Good point. updated